### PR TITLE
Added …Inlet.create Java API for errorHandler.

### PR DESCRIPTION
**Pull request by RayRoestenburg**
### What changes were proposed in this pull request?

Added Java API for ProtoInlet and AvroInlet, for errorHandler.

### CODE REVIEW

File: core/cloudflow-streamlets/src/main/scala/cloudflow/streamlets/avro/AvroInlet.scala [hunk 1]
```diff
@@ -40,6 +41,18 @@ object AvroInlet {
...
+
+  // Java API
+  def create[T <: SpecificRecordBase](
+      name: String,
+      clazz: Class[T],
+      hasUniqueGroupId: Boolean,
+      errorHandler: (Array[Byte], Throwable) => Optional[T]): AvroInlet[T] =
+    AvroInlet[T](name, hasUniqueGroupId, (a, t) => {
+      val opt = errorHandler(a, t)
+      if (opt.isPresent) Some(opt.get()) else None
```
Comment (debasishg) [inline core/cloudflow-streamlets/src/main/scala/cloudflow/streamlets/avro/AvroInlet.scala:56]: Isn't this same as returning the `opt` itself ?

File: core/cloudflow-streamlets/src/main/scala/cloudflow/streamlets/proto/javadsl/ProtoInlet.scala [hunk 2]
```diff
@@ -39,10 +39,19 @@ final case class ProtoInlet[T <: GeneratedMessageV3](
...
     ProtoInlet[T](name, clazz, hasUniqueGroupId)
+
+  def create[T <: GeneratedMessageV3](
+      name: String,
+      clazz: Class[T],
+      hasUniqueGroupId: Boolean,
+      errorHandler: (Array[Byte], Throwable) => Optional[T]): ProtoInlet[T] =
+    ProtoInlet[T](name, clazz, hasUniqueGroupId, (a, t) => {
+      val opt = errorHandler(a, t)
+      if (opt.isPresent) Some(opt.get()) else None
```
Comment (debasishg) [inline core/cloudflow-streamlets/src/main/scala/cloudflow/streamlets/proto/javadsl/ProtoInlet.scala:55]: Isn't this same as returning the opt itself ?

**Review by debasishg — APPROVED** [review]
a minor suggestion .. otherwise LGTM

File: core/cloudflow-streamlets/src/main/scala/cloudflow/streamlets/avro/AvroInlet.scala (hunk 1 above)
Comment (RayRoestenburg) [reply core/cloudflow-streamlets/src/main/scala/cloudflow/streamlets/avro/AvroInlet.scala:56]: LOL I must have refactored this somehow.. good catch!

File: core/cloudflow-streamlets/src/main/scala/cloudflow/streamlets/proto/javadsl/ProtoInlet.scala (hunk 2 above)
Comment (RayRoestenburg) [reply core/cloudflow-streamlets/src/main/scala/cloudflow/streamlets/proto/javadsl/ProtoInlet.scala:55]: Oh no, wait, actually this is converted to the Scala API. The constructor takes Option, not Optional.

File: core/cloudflow-streamlets/src/main/scala/cloudflow/streamlets/avro/AvroInlet.scala (hunk 1 above)
Comment (RayRoestenburg) [reply core/cloudflow-streamlets/src/main/scala/cloudflow/streamlets/avro/AvroInlet.scala:56]: It's calling the Scala constructor which needs `Option`, not `Optional`

**Pull request merged**